### ISSUE TITLE
authorization empty to context

### ISSUE THREAD

**Issue by luizcarlospedrosogomes** [cap-js/docs#1177]
```js
this.on('READ', 'Something', async req => {
    const svc = await cds.connect.to('AnotherService');
    const res = await svc.run(SELECT.from('RemoteThing', someId));
    return res;
})
```

```js
 // IMPORTANT: regular function is used on purpose, don't switch to arrow function.
    this.on('*', async function on_handler(req, next) {
      const { query } = req
      if (!query && !(typeof req.path === 'string')) return next()

      // early validation on first request for use case without remote API
      // ideally, that's done on bootstrap of the remote service
      if (typeof this.destination === 'object' && !this.destination.url)
        throw new Error(`"url" or "destination" property must be configured in "credentials" of "${this.name}".`)

      const reqOptions = getReqOptions(req, query, this)
      reqOptions.headers = _setHeaders(reqOptions.headers, req)

      const { kind, destination, destinationOptions } = this
      const resolvedTarget = resolvedTargetOfQuery(query) || getTransition(req.target, this).target
      const returnType = req._returnType
      const additionalOptions = { destination, kind, resolvedTarget, returnType, destinationOptions }

      // req.context.headers only contains 'x-correlation-id' after the upgrade, no jwt token is forwarded
      // const jwt = req?.context?.headers?.authorization?.split(/^bearer /i)[1]
      const jwt = req?.context?.http?.req?.headers?.authorization?.split(/^bearer /i)[1] // ---> Changing like this makes it work?
      if (jwt) additionalOptions.jwt = jwt

      // hidden compat flag in order to suppress logging response body of failed request
      if (req._suppressRemoteResponseBody) {
        additionalOptions.suppressRemoteResponseBody = req._suppressRemoteResponseBody
      }

      let result = await run(reqOptions, additionalOptions)
      result = typeof query === 'object' && query.SELECT?.one && Array.isArray(result) ? result[0] : result
      return result
    })

```

req?.context?.headers?.authorization is empty

**Comment by carlonnheim** [comment]
Hi @chgeo ,

This started as a community post here, with some more elaboration on it. This stopped working after CDS8 upgrade.

https://community.sap.com/t5/technology-q-a/cap-cds8-remote-services-with-principal-propagation-fails/qaq-p/13781552

Thanks!

**Comment by chgeo** [comment]
@sjvans @johannes-vogel @David-Kunz can someone in your team take of the issue?
Feel free to close this issue here in favor of the community post.

**Comment by sjvans** [comment]
hi @carlonnheim @chgeo 

this was fixed in @sap/cds@8.1.1.

best,
sebastian
